pkg/process/metadata/parser/service.go

```diff
@@ -217,6 +231,32 @@ func isRuneLetterAt(s string, position int) bool {
 	return len(s) > position && unicode.IsLetter(rune(s[position]))
 }
 
+// stripAndScanEnvs scans the cmd args and remove the env variable definition at the beginning.
+// If the service name has been defined through a standard env variable (DD_SERVICE or DD_TAGS) it returns it.
+// Otherwise, it modifies the cmd arg by skipping all the envs
+func stripAndScanEnvs(cmd *[]string) (string, bool) {
+	pos := 0
+	for i, arg := range *cmd {
+		if strings.ContainsRune(arg, '=') {
+			pos = i + 1
+			if strings.HasPrefix(arg, "DD_SERVICE=") {
+				return strings.TrimPrefix(arg, "DD_SERVICE="), true
+			} else if strings.HasPrefix(arg, "DD_TAGS=") {
+				parts := strings.Split(strings.TrimPrefix(arg, "DD_TAGS="), ",")
+				for _, p := range parts {
+					if strings.HasPrefix(p, "service:") {
+						return strings.TrimPrefix(p, "service:"), true
+					}
+				}
+			}
+		} else {
+			break
+		}
```

Comment:
Redundant nesting that can be easily tweaked
```suggestion
      if !strings.ContainsRune(arg, '=') {
          break
      }
      pos = i + 1
      if strings.HasPrefix(arg, "DD_SERVICE=") {
          return strings.TrimPrefix(arg, "DD_SERVICE="), true
      } else if strings.HasPrefix(arg, "DD_TAGS=") {
          parts := strings.Split(strings.TrimPrefix(arg, "DD_TAGS="), ",")
          for _, p := range parts {
	          if strings.HasPrefix(p, "service:") {
		          return strings.TrimPrefix(p, "service:"), true
	          }
          }
      }
```